anndata/readwrite/write.py

```diff
@@ -4,14 +4,24 @@
 import pandas as pd
 import math
 import numpy as np
-from scipy.sparse import issparse
+from scipy import sparse
 import logging as logg
+import zarr
 
 from ..base import AnnData
 from .. import h5py
 from ..compat import PathLike, fspath
 
 
+def issparse(value):
```

Comment:
This function is now being used where scipy.sparse. issparse was being used in some h5 codepaths. Is that expected?

Reply:
yea, similar to the above, the HDF5-backed AnnData I was plumbing through here does not register as `issparse` according to `scipy`, and that was evidently the source of a lot of woes. The only way I saw to detect that this was in fact a sparse dataset was this `format_str == 'csr'`

I'll add some tests of all the relevant paths here